### DISCUSSION TITLE
VT100 double-height, etc

### DISCUSSION THREAD

**Discussion by dgl** [ghostty-org/ghostty#3243] [Feature Requests, Ideas]
Apple's built-in terminal supports double height VT100 escape sequences:

```
printf '\e#3Hello world ^ 🥳\n\e#4Hello world : 😝\n'
```

<img width="601" alt="Image" src="https://github.com/user-attachments/assets/30bcde0e-ebbb-4ac9-b921-a432b8f3927e" />

It's useful for printing titles and so on and works on most terminals. While clearly the VT100 didn't support emoji, several terminals implement this as an obvious extension and you can even use it to invent new emoji combinations for great fun (see above). This works in Apple terminal and xterm (although there were some bugs I reported and were fixed around 2 years ago). (I also have a public thing which uses this if you run `curl -i ip.wtf`.)

**Comment by 00-kat** [comment]
Related: #5563.

**Comment by glaforge** [comment]
There was a recent article on this: https://dgl.cx/2025/06/can-your-terminal-do-emojis
It reminded me of this feature that I used sometimes in macOS' terminal.

I was going to create a ticket, but found this discussion that already existed.

Could we support this double height feature from VT100?
Shall I create a feature request ticket to track its progress?

**Reply by 00-kat** [reply]
> Shall I create a feature request ticket to track its progress?

This *is* the feature request though.

**Comment by wiesmann** [comment]
For the reference, this is the full list of escape sequences, taken from [here](https://espterm.github.io/docs/VT100%20escape%20codes.html)

| name | DEC         | Description                                          | Sequence |
|-------|----------|-------------------------------------|-----------|
|dhtop  | DECDHL  |  Double-height letters, top half        |    ^[#3      |
| dhbot | DECDHL  |  Double-height letters, bottom half |    ^[#4       |
| swsh  | DECSWL  |  Single width, single height letters   |   ^[#5        |
| dwsh  | DECDWL |  Double width, single height letters  |   ^[#6        |

Besides for titles, I use this to zoom in for complicated characters which are difficult to read at normal terminal sizes, like for instance 龘.
